src/librustc/traits/util.rs

```diff
@@ -0,0 +1,17 @@
+use crate::ty::TyCtxt;
+use rustc_hir as hir;
+use rustc_hir::def_id::DefId;
+
+pub fn impl_is_default(tcx: TyCtxt<'_>, node_item_def_id: DefId) -> bool {
+    match tcx.hir().as_local_hir_id(node_item_def_id) {
```

Comment:
This sort of code (one branch for local and one for cross-crate) shouldn't really exist nowadays in the query world.